File: packages/audio/android/src/main/java/dev/tcode/thinmpr/audio/model/AlbumModel.kt
package dev.tcode.thinmpr.audio.model

import android.net.Uri
import dev.tcode.thinmpr.audio.constant.MediaConstant
import dev.tcode.thinmpr.audio.model.contract.AlbumModelContract
import dev.tcode.thinmpr.audio.model.valueObject.AlbumId
import dev.tcode.thinmpr.audio.model.valueObject.ArtistId

class AlbumModel(
    id: String,
    name: String,
    artistId: String,
    artistName: String
): AlbumModelContract {
    override val id: AlbumId = AlbumId(id)
    override val name: String = name
    override val artistId: ArtistId = ArtistId(artistId)
    override val artistName: String = artistName

    override val imageId: String
        get() = id.raw

    override val imageUri: Uri
        get() =  Uri.parse("${MediaConstant.ALBUM_ART}/${id.raw}")
}

File: model/AlbumModel.ts
import { AlbumDTO } from "audio";

export class AlbumModel {
  id: string;
  name: string;
  artistId: string;
  artistName: string;
  imageId: string;

  constructor(
    id: string,
    name: string,
    artistId: string,
    artistName: string,
    imageId: string,
  ) {
    this.id = id;
    this.name = name;
    this.artistId = artistId;
    this.artistName = artistName;
    this.imageId = imageId;
  }

  static fromDTO(album: AlbumDTO): AlbumModel {
    return new AlbumModel(
      album.id,
      album.name,
      album.artistId,
      album.artistName,
      album.imageId,
    );
  }
}
